### ISSUE TITLE
[inductor] `max` returns incorrect indices when compiling a function includes `(a + b).max(dim=1)`

### ISSUE THREAD

**Issue by maybeLee** [pytorch/pytorch#178964]
### 🐛 Describe the bug

The following operator sequence trigger this inconsistency between torch.compile and eager: `(a + b).max(dim=1)`, where `a` is a transposed tensor. Specifically, the returned indices of `max` operation are all zeros instead of correct indices.

Here is the code to reproduce:
```python
import torch
def fn(a, b):
    return (a + b).max(dim=1)

torch.manual_seed(0)
a = torch.randn(8, 8).transpose(0, 1)  # the shape should be (8, 8) to trigger the error
b = torch.randn(8, 8)

res1 = fn(a, b)
print(res1.indices)  # tensor([5, 5, 6, 7, 0, 3, 7, 2])
res2 = torch.compile(fn, backend='inductor')(a, b)
print(res2.indices)  # tensor([0, 0, 0, 0, 0, 0, 0, 0])
torch.testing.assert_close(res1, res2, equal_nan=True)  # assertion error

```

Interestingly, if we execute `a+b` outside of the compiled function, no error occurs:

```python

# This works fine
def fn2(x):
    return x.max(dim=1)
res1 = fn2(a+b)
print(res1.indices)  # tensor([5, 5, 6, 7, 0, 3, 7, 2])
res2 = torch.compile(fn2)(a+b)
print(res2.indices)  # tensor([5, 5, 6, 7, 0, 3, 7, 2])
torch.testing.assert_close(res1, res2, equal_nan=True)  # works fine
```

This issue happens when using backend `inductor`, and does not happen when using backend `aot_eager` or `eager`.

### Versions

[pip3] numpy==2.4.2
[pip3] torch==2.12.0.dev20260328+cpu
[pip3] torchvision==0.26.0.dev20260223+cpu
[pip3] triton==3.6.0
[conda] numpy 2.4.2 pypi_0 pypi
[conda] torch 2.12.0.dev20260328+cpu pypi_0 pypi
[conda] torchvision 0.26.0.dev20260223+cpu pypi_0 pypi
[conda] triton 3.6.0 pypi_0 pypi

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo
